// backend/dist/seeder.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const dotenv_1 = __importDefault(require("dotenv"));
const Project_1 = __importDefault(require("./models/Project"));
const Skill_1 = __importDefault(require("./models/Skill"));
const User_1 = __importDefault(require("./models/User"));
const Settings_1 = __importDefault(require("./models/Settings"));
const db_1 = __importDefault(require("./config/db"));
dotenv_1.default.config();
const seedData = async () => {
    try {
        await (0, db_1.default)();
        // Clear existing data
        await Project_1.default.deleteMany();
        await Skill_1.default.deleteMany();
        await User_1.default.deleteMany();
        await Settings_1.default.deleteMany();
        // Create Admin User
        const adminUser = new User_1.default({
            username: process.env.ADMIN_USERNAME || 'admin',
            password: process.env.ADMIN_PASSWORD || 'password123',
        });
        await adminUser.save();
        // Create Site Settings
        await Settings_1.default.create({
            siteName: 'ShamVerse',
            heroTitle: 'Crafting Digital Universes at ShamVerse',
            heroSubtitle: 'Full-Stack Developer',
            bio: 'Professional full-stack developer specializing in building high-performance, visually stunning web applications.',
        });
        // Seed Skills
        const skills = [
            { name: 'React', category: 'Frontend', level: 95 },
            { name: 'TypeScript', category: 'Frontend', level: 90 },
            { name: 'Node.js', category: 'Backend', level: 85 },
            { name: 'MongoDB', category: 'Database', level: 80 },
            { name: 'Framer Motion', category: 'Animation', level: 90 },
        ];
        await Skill_1.default.insertMany(skills);
        // Seed Projects
        const projects = [
            {
                title: 'ShamVerse Portfolio',
                description: 'A full-stack developer portfolio with a custom dark maroon theme and advanced animations.',
                image: 'https://via.placeholder.com/600x400',
                liveLink: 'https://shamverse.com',
                githubLink: 'https://github.com/sham/shamverse',
                tags: ['React', 'Node.js', 'MongoDB', 'Framer Motion'],
                featured: true,
            },
            {
                title: 'E-Commerce Universe',
                description: 'A high-performance e-commerce platform with real-time inventory management.',
                image: 'https://via.placeholder.com/600x400',
                liveLink: 'https://shop.shamverse.com',
                githubLink: 'https://github.com/sham/ecommerce',
                tags: ['Next.js', 'Stripe', 'Tailwind'],
                featured: false,
            }
        ];
        await Project_1.default.insertMany(projects);
        console.log('Data Seeded Successfully!');
        process.exit();
    }
    catch (error) {
        console.error('Error seeding data:', error);
        process.exit(1);
    }
};
seedData();
